src/go/util/string.go

```diff
@@ -19,3 +21,32 @@ func RandomString(n int) string {
 
 	return string(b)
 }
+
+// Credit: https://github.com/go-yaml/yaml/issues/100#issuecomment-901604971
+//
+// Allow yaml parsing for an array of strings or a single inlined element:
+//
+//	field: value
+//
+//	  or
+//
+//	field:
+//	  - value1
+//	  - value2
+type StringArray []string
+
+func (a *StringArray) UnmarshalYAML(value *yaml.Node) error {
+	var multi []string
+	err := value.Decode(&multi)
+	if err != nil {
+		var single string
+		err := value.Decode(&single)
+		if err != nil {
+			return err
+		}
+		*a = []string{single}
+	} else {
+		*a = multi
+	}
+	return nil
+}
```

Comment:
Will likely need to have a custom implementation of `UnmarshalJSON` as well.

Reply:
Without any additional changes, I tested with:

```bash
phenix cfg get topology/foobar -o json -p > foobar.json
phenix cfg delete topology/foobar
phenix cfg create /phenix/topologies/foobar.json
```

Output was: `The Topology/foobar configuration was created`, and the yaml shown with `cfg get` was correct.

Where else should I check?

Reply:
`topology/foobar` had 2 vms: one with dns as a string, one as an array with 2 entries.

Reply:
@glattercj getting the JSON from phenix will output an array of DNS names. Did you test creating from JSON when the DNS entry was a string instead of an array of strings?

Reply:
> `topology/foobar` had 2 vms: one with dns as a string, one as an array with 2 entries.

Was the DNS entry for the one VM a string when you ran `cfg create` using the JSON file (your last step)? When you export from phenix to JSON, it will be an array no matter what since it's a string slice in the topology schema.

Reply:
[foo-yaml.md](https://github.com/sandia-minimega/phenix/files/10492626/foo-yaml.md)
[foo-json.md](https://github.com/sandia-minimega/phenix/files/10492625/foo-json.md)

Reply:
Getting from phenix the json matched: one vm was a string, one was an array with 2 entries. It seems to be an array when it gets added as an experiment, then the exp config shows both as arrays.

Reply:
Cool! Looks like the JSON parser handles that situation automatically. 👍🏻👍🏻

Reply:
@glattercj I'm assuming you added the custom yaml unmarshaller because it wouldn't work by default?

Reply:
@glattercj I just did some testing and you don't need the custom `StringSlice` type with the custom `UnmarshalYAML` function. I think everything works fine now that you updated the schema to allow for string or array.

Reply:
Fixed